app/src/main/java/com/github/se/icebreakrr/model/profile/ProfilesViewModel.kt

```diff
@@ -269,34 +284,84 @@ open class ProfilesViewModel(
     _tempProfilePictureBitmap.value = bitmap
   }
 
+  /**
+   * Saves the edited profile to the state.
+   *
+   * @param profile The profile to be saved.
+   */
+  fun saveEditedProfile(profile: Profile) {
+    _editedCurrentProfile.value = profile
+  }
+
   /** Clears the temporary profile picture Bitmap. */
   fun clearTempProfilePictureBitmap() {
     _tempProfilePictureBitmap.value = null
   }
 
+  /** Clears the edited profile from the state. */
+  private fun clearEditedProfile() {
+    _editedCurrentProfile.value = null
+  }
+
+  /** Resets every state related to profile edition. */
+  fun resetProfileEditionState() {
+    clearTempProfilePictureBitmap()
+    clearEditedProfile()
+    _pictureChangeState.value = ProfilePictureState.UNCHANGED
+  }
+
   /**
    * Processes the cropped image by compressing it if it is too large.
    *
    * @param bitmap The cropped Bitmap to be processed.
    */
   fun processCroppedImage(bitmap: Bitmap) {
     _tempProfilePictureBitmap.value = compressImageIfTooLarge(bitmap)
+    _pictureChangeState.value = ProfilePictureState.TO_UPLOAD
+  }
+
+  fun validateProfileChanges(context: Context) {
+    when (_pictureChangeState.value) {
+      ProfilePictureState.TO_UPLOAD ->
+          validateAndUploadProfilePicture(context) { url ->
+            val newProfile = _editedCurrentProfile.value!!.copy(profilePictureUrl = url)
+            updateProfile(newProfile)
+            _selfProfile.value = newProfile
+            // todo: "_selectedProfile.value = newProfile" needs to be removed when all the other
+            // personal profile views are updated to use the selfProfile instead of the
+            // selectedProfile
+            _selectedProfile.value = newProfile
+
+            resetProfileEditionState()
+          }
+      ProfilePictureState.TO_DELETE ->
+          deleteCurrentUserProfilePicture() {
+            val newProfile = _editedCurrentProfile.value?.copy(profilePictureUrl = null)
+            updateProfile(newProfile!!)
+            _selfProfile.value = newProfile
+            _selectedProfile.value = newProfile // todo: same as above
+            resetProfileEditionState()
+          }
+      else -> {
+        updateProfile(_editedCurrentProfile.value!!)
+        _selfProfile.value = _editedCurrentProfile.value
+        _selectedProfile.value = _editedCurrentProfile.value // todo: same as above
+        resetProfileEditionState()
+      }
+    }
   }
 
   /**
    * Validates and uploads the profile picture if a temporary profile picture Bitmap exists.
    *
    * @param context The context used to show a Toast message in case of failure.
    */
```

Comment:
There is a lack of documentation here, the parameter onSucces isn't described

Reply:
OUCH... you didn't read my last message on discord saying to wait 5m for me to update the outdated doc. You started reviewing 30s before the push was made leading you to review an outdated code. That one's on me, I should have been more clear. I'll marke the other comment's with an emoji